ext/php8/serializer.c

```diff
@@ -426,6 +426,36 @@ static void _serialize_meta(zval *el, ddtrace_span_fci *span_fci) {
         add_assoc_string(meta, "system.pid", pid);
     }
 
+    const char *version = get_dd_version();
+    if (*version) {  // non-empty
+        add_assoc_string(meta, "version", version);
+    }
+
+    const char *env = get_dd_env();
+    if (*env) {  // non-empty
+        add_assoc_string(meta, "env", env);
+    }
+
+    zend_array *global_tags = get_dd_tags();
+    if (zend_hash_num_elements(global_tags) == 0) {
+        zend_hash_release(global_tags);
+        global_tags = get_dd_trace_global_tags();
+    }
+    zend_string *global_key;
+    zval *global_val;
+    ZEND_HASH_FOREACH_STR_KEY_VAL(global_tags, global_key, global_val) {
+        zend_hash_add(Z_ARR_P(meta), global_key, global_val);
+    }
+    ZEND_HASH_FOREACH_END();
+    zend_hash_release(global_tags);
+
+    zend_string *tag_key;
+    zval *tag_value;
+    ZEND_HASH_FOREACH_STR_KEY_VAL(DDTRACE_G(additional_global_tags), tag_key, tag_value) {
+        zend_hash_add(DDTRACE_G(additional_global_tags), tag_key, tag_value);
```

Comment:
Not sure I understand what this line does. I think that here we should add from `DDTRACE_G(additional_global_tags)` into `Z_ARR_P(meta)`?

If this holds true, then we are missing a test for global + additional globals?

Reply:
Yes - and yes, for sure that'll get a test.